### ISSUE TITLE
Bug with xml field type SQL

### ISSUE THREAD

**Issue by sdi68** [joomla/joomla-cms#22859]
### Steps to reproduce the issue
I need to use field type sql with param sql_filter (see https://docs.joomla.org/SQL_form_field_type)

### Expected result
I query added condition with values in this field. 

### Actual result
Condition not added.

### System information (as much as possible)
Joomla 3.x

### Additional comments
I fix this with edit the file: \libraries\joomla\form\fields\sql.php.
![screen shot 2018-10-30 at 09 04 47](https://issues.joomla.org/uploads/1/591744945ab83e22941bc313a2a01c0d.png)My code add with comments //SDI.

**Comment by Quy** [comment]
Is this related #22241? Please provide an XML example in your case.

**Comment by sdi68** [comment]
This is code example from my case			
<fieldset name="basic">

				<field
					name="catid"
					type="category"
					extension="com_content"
					multiple="false"
					default=""
					size="10"
					label="JCATEGORY"
					description="MOD_SDI_SELART_CATEGORY_DESC"
				>
					<option value="">JOPTION_ALL_CATEGORIES</option>
				</field>

				<field
						name="artids"
						type="sql"
						multiple="true"
						default=""
						size="10"
						label="JGLOBAL_ARTICLES"
						description="MOD_SDI_SELART_ARTICLES_DESC"
						sql_select="id, title"
						sql_from="#__content"
						sql_filter= "catid"
						sql_where="state=1"
						key_field="id"
						value_field="title"
						sql_default_catid=""
				>
				</field>
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22859">issues.joomla.org/tracker/joomla-cms/22859</a>.</sub>

**Comment by sdi68** [comment]
I try this from module.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22859">issues.joomla.org/tracker/joomla-cms/22859</a>.</sub>

**Comment by jbrailas** [comment]
I can reproduce this problem but your code doesn't work in order to solve this bug.

**Comment by sdi68** [comment]
> I can reproduce this problem but your code doesn't work in order to solve this bug.

Sorry! I forgot add second part of fix. In the same file (see comment SDI)
![2019-01-15_18-18-02](https://user-images.githubusercontent.com/36002964/51189974-42993580-18f2-11e9-9ed2-4ec7015e0889.png)

**Comment by jbrailas** [comment]
It is ok now. But the first part of your code should be:
```
$curfrm = $this->form;
if (is_array($filters)) :
	foreach ($filters as $i => $v)
	{
		$filterval = $curfrm->getValue ($v, $group);
		if (!is_null($filterval)) 
			$filters[$i]= $v . '=' . $filterval;
	}
endif;
```
If you don't add "if (is_array($filters)) " then you will have a warning: Invalid argument supplied for foreach() in other sql fields, which don't use the "sql-filter".

Someone should include this fix in the next version of joomla.
However, it doesn't work dynamically, meaning it doesn't change the values after the page is loaded, so it is not very useful.

**Comment by fastslack** [comment]
@sdi68 Try with this PR https://github.com/joomla/joomla-cms/pull/24755

**Comment by ghost** [comment]
Closed as having Pull Request #24755. Please reopen if PR didn't fix the Issue.

**Comment by jbrailas** [comment]
Why the solution is not implemented in Joomla core file sql.php yet?
